core/include/gnuradio-4.0/Settings.hpp

```diff
@@ -44,37 +67,45 @@ hash_combine(std::size_t &seed, const T &v) noexcept {
 
 struct SettingsCtx {
     // using TimePoint = std::chrono::time_point<std::chrono::utc_clock>; // TODO: change once the C++20 support is ubiquitous
-    using TimePoint               = std::chrono::time_point<std::chrono::system_clock>;
-    std::optional<TimePoint> time = std::nullopt; /// UTC time-stamp from which the setting is valid
-    property_map             context;             /// user-defined multiplexing context for which the setting is valid
+    using TimePoint   = std::chrono::time_point<std::chrono::system_clock>;
+    uint64_t  time    = 0ULL; // UTC-based time-stamp in ns, time from which the setting is valid, 0U is undefined time
+    pmtv::pmt context = "";   // user-defined multiplexing context for which the setting is valid
 
     SettingsCtx() {}
 
-    explicit SettingsCtx(const TimePoint &t, const property_map &ctx = {}) {
+    explicit SettingsCtx(uint64_t t, const pmtv::pmt &ctx = "") {
         time    = t;
         context = ctx;
     }
 
+    explicit SettingsCtx(const TimePoint &tp, const pmtv::pmt &ctx = "") {
```

Comment:
Please remove this constructor. functionally identical to the one above and widens the public API by a non-Python wrappable constructor.

Actually, do we need a dedicated constructor? :thinking:

Reply:
It was used  to facilitate unit testing. I removed it.